bin/metric-zfs-arc.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+#
+#   ZFS ARC metrics
+#
+# DESCRIPTION:
+#   This collects ZFS ARC (Adaptive Replacement Cache) metrics
+#   metric-zfs-arc.rb looks at /proc/spl/kstat/zfs/arcstats
+#
+# OUTPUT:
+#  metric data
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: socket
+#
+# USAGE:
+#  ./metric-zfs-arc.rb
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2019 Airbrake Technologies, Inc <support@airbrake.io>
+#   Released under the same terms as the Sensu (the MIT license); see LICENSE
+#   for details.
+
+require 'sensu-plugin/metric/cli'
+require 'socket'
+
+#
+# ZFS Arc Metrics
+##
+class ZfsArcMetrics < Sensu::Plugin::Metric::CLI::Graphite
+  option :scheme,
+         description: 'Metric naming scheme, text to prepend to .$parent.#child',
+         short: '-s SCHEME',
+         long: '--scheme SCHEME',
+         default: "#{Socket.gethostname}.zfs"
+
+  def run
+    if File.exist?('/proc/spl/kstat/zfs/arcstats')
+      command = 'cat /proc/spl/kstat/zfs/arcstats'
+
+      `#{command}`.split("\n").drop(2).each do |row|
+        name, _type, data = row.split
```

Comment:
you might want to check to make sure the row is not nil before splitting.
```
if row.nil? 
  # some error handling
else
  name, _type, data = row.split
end
```

Reply:
> you might want to check to make sure the row is not nil before splitting.

This makes sense.

I wonder if we can't use the [safe navigation operator](calling_methods_rdoc.html#label-Safe+navigation+operator) instead

```
name, _type, data = row&.split
output "#{config[:scheme]}.#{name}", data
```

Reply:
In order to use safe navigation we need to drop ruby version support for `< 2.3`: https://github.com/sensu-plugins/sensu-plugins-zfs/blob/master/sensu-plugins-zfs.gemspec#L35

Reply:
@majormoses no worries. I'll update https://github.com/sensu-plugins/sensu-plugins-zfs/pull/21

Reply:
I suspect we can do the cleanup in a follow-up PR after #21 is merged.